src/main/java/Event.java

```diff
@@ -0,0 +1,48 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
+
+    protected String at;
+    protected LocalDate formattedDate;
+
+    /**
+     * Constructor for Event class
+     *
+     * @param description Description for event.
+     * @param at          Date of event
+     * @throws DukeException If date/description not specified.
+     */
+    public Event(String description, String at) throws DukeException {
+        super(description);
+        if (at.equalsIgnoreCase("")) {
+            throw new DukeException("No date");
```

Comment:
I like that you made your own exception and used it but I feel that there can be more! For example having different exceptions for different reasons so that you don't do a so-called Pokemon exception handling.